modules/mono/glue/GodotSharp/GodotSharp/Core/MarshalUtils.cs

```diff
@@ -16,8 +16,8 @@ static class MarshalUtils
         /// <exception cref="System.InvalidOperationException">
         /// <paramref name="type"/> is not a generic type. That is, IsGenericType returns false.
         /// </exception>
-        static bool TypeIsGenericArray(Type type) =>
-            type.GetGenericTypeDefinition() == typeof(Godot.Collections.Array<>);
+        private static bool TypeIsGenericArray(Type type) =>
+            type.GetGenericTypeDefinition() == typeof(Collections.Array<>);
```

Comment:
Can this just be `typeof(Array<>)` because we have `using Array = Godot.Collections.Array`?

Reply:
I prefer to be very specific in marshaling code to avoid confusion, so I prefer to always specify Array and Dictionary namespaces. Also I just re-read the file and the usings cause a lot of confusion because they don't apply to the generic versions. I will remove them. For a moment I thought `TypeIsSystemGenericDictionary` was incorrectly changed and was now checking for Godot's generic dict instead of System's.